### DISCUSSION TITLE
Ory Hydra 2.0 and MySQL 5.7

### DISCUSSION THREAD

**Discussion by BrandonNoad** [ory/hydra#3357] [Q&A] [answered]
Is Hydra 2.0 compatible with MySQL 5.7? 

https://www.ory.sh/docs/hydra/self-hosted/dependencies-environment#database-configuration seems to indicate that it is.

When running the migrations, I am getting an error executing the migrations in the following [file](https://github.com/ory/hydra/blob/5f5c81ea0883d83d5d1f6a52cca9c3a7148adfd8/persistence/sql/migrations/20211019000001000000_merge_authentication_request_tables.mysql.up.sql), `migrations/20211019000001000000_merge_authentication_request_tables.mysql.up.sql`.

https://dev.mysql.com/doc/refman/5.7/en/data-type-defaults.html

> The [BLOB](https://dev.mysql.com/doc/refman/5.7/en/blob.html), [TEXT](https://dev.mysql.com/doc/refman/5.7/en/blob.html), GEOMETRY, and [JSON](https://dev.mysql.com/doc/refman/5.7/en/json.html) data types cannot be assigned a default value.

The migration file above has default values for `text` columns, which is not supported in MySQL 5.7.

**Comment by aeneasr** [comment]
We should probably bump that requirement to MySQL 8.0 in the docs!

**Comment by BrandonNoad** [comment, marked as answer]
So I guess the answer is no.

The CHANGELOG says:

> We removed compatibility with unsupported database versions (e.g. MySQL 5.6). Ory Hydra v2.x is now compatible with MySQL 8.0.13+, PostgreSQL 11.8+, CockroachDB v22.1.2+.

That makes it sound like MySQL 5.6 is no longer supported. MySQL >= 8.0.13 is now supported. And MySQL 5.7 support is unchanged. So you may want to update that as well.

**Reply by aeneasr** [reply]
True, I though

> Ory Hydra v2.x is now compatible with MySQL 8.0.13+, PostgreSQL 11.8+, CockroachDB v22.1.2+.

is bringing the message across, but I guess we should explicitly have added MySQL 5.7 to the list of deprecation.
